v1/transform.go

```diff
@@ -0,0 +1,228 @@
+			Description: r.Description,
+			Tags:        toV1Tags(r.Tags),
+			Attributes:  toV1Attribute(r.Attributes),
+			Properties: v1Models.ProfileProperty{
+				Value: v1Models.PropertyValue{
+					Type:          r.Properties.ValueType,
+					ReadWrite:     r.Properties.ReadWrite,
+					Minimum:       r.Properties.Minimum,
+					Maximum:       r.Properties.Maximum,
+					DefaultValue:  r.Properties.DefaultValue,
+					Mask:          r.Properties.Mask,
+					Shift:         r.Properties.Shift,
+					Scale:         r.Properties.Scale,
+					Offset:        r.Properties.Offset,
+					Base:          r.Properties.Base,
+					Assertion:     r.Properties.Assertion,
+					FloatEncoding: "eNotation",
+					MediaType:     r.Properties.MediaType,
+				},
+				Units: v1Models.Units{
+					Type:         "String",
+					ReadWrite:    common.ReadWrite_RW,
+					DefaultValue: r.Properties.Units,
+				},
+			},
+		}
+	}
+
+	return resources
+}
+
+func toV1Tags(tags map[string]interface{}) map[string]string {
+	dto := make(map[string]string)
+	for k, v := range tags {
+		dto[k] = fmt.Sprintf("%v", v)
+	}
+	return dto
+}
+
+func toV1Attribute(attributes map[string]interface{}) map[string]string {
+	dto := make(map[string]string)
+	for k, v := range attributes {
+		dto[k] = fmt.Sprintf("%v", v)
+	}
+	return dto
+}
+
+const (
+	ResourceOperationGet = "get"
+	ResourceOperationSet = "set"
+)
+
+func toV1DeviceCommands(deviceCommands []models.DeviceCommand) []v1Models.ProfileResource {
+	commands := make([]v1Models.ProfileResource, len(deviceCommands))
+	for i, c := range deviceCommands {
+		commands[i] = v1Models.ProfileResource{
+			Name: c.Name,
+		}
+		if strings.Contains(c.ReadWrite, common.ReadWrite_R) {
+			commands[i].Get = toV1Operation(ResourceOperationGet, c.ResourceOperations)
+		}
+		if strings.Contains(c.ReadWrite, common.ReadWrite_W) {
+			commands[i].Set = toV1Operation(ResourceOperationSet, c.ResourceOperations)
+		}
+	}
+
+	return commands
+}
+
+func toV1Operation(op string, resourceOperations []models.ResourceOperation) []v1Models.ResourceOperation {
+	operations := make([]v1Models.ResourceOperation, len(resourceOperations))
+	for i, ro := range resourceOperations {
+		operations[i] = v1Models.ResourceOperation{
+			Index:          strconv.Itoa(i),
+			Operation:      op,
+			DeviceResource: ro.DeviceResource,
+			Mappings:       ro.Mappings,
+		}
+	}
+	return operations
+}
+
+// TransformToV2DeviceProfile transform v1 profile to v2
+func TransformToV2DeviceProfile(profile v1Models.DeviceProfile) models.DeviceProfile {
+	dto := models.DeviceProfile{
+		Description:     profile.Description,
+		Name:            profile.Name,
+		Manufacturer:    profile.Manufacturer,
+		Model:           profile.Model,
+		Labels:          profile.Labels,
+		DeviceResources: nil,
+		DeviceCommands:  nil,
+	}
+	dto.DeviceResources = toV2DeviceResources(profile.DeviceResources)
+	dto.DeviceCommands = toV2DeviceCommands(profile.DeviceCommands)
+	return dto
```

Comment:
use CoreCommand to determine the value of isHidden

Reply:
Added func to check whether the resource exists in the CoreCommands or DeviceCommands, if exists, the resource is not hidden.

Reply:
isHidden only be impacted by CoreCommand, not DeviceCommand.
isHidden of DeviceCommand is also accordance with CoreCommand.

Reply:
Modified. 
- Set isHidden to true as default for v2DeviceResource and v2DeviceCommand
- then determine the isHidden value by CoreCommand